<h5>Importing libraries</h5>

In [2]:
import pandas as pd
import numpy as np
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')

<h3>Importing dataset and converting it into column</h3>

In [3]:
uk = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/datasets/uk_road_accident.csv')

In [4]:
uk

Index Accident_Severity Accident Date   Latitude  \
0       200701BS64157           Serious      5/6/2019  51.506187   
1       200701BS65737           Serious      2/7/2019  51.495029   
2       200701BS66127           Serious    26-08-2019  51.517715   
3       200701BS66128           Serious    16-08-2019  51.495478   
4       200701BS66837            Slight      3/9/2019  51.488576   
...               ...               ...           ...        ...   
660674  201091NM01760            Slight    18-02-2022  57.374005   
660675  201091NM01881            Slight    21-02-2022  57.232273   
660676  201091NM01935            Slight    23-02-2022  57.585044   
660677  201091NM01964           Serious    23-02-2022  57.214898   
660678  201091NM02142           Serious    28-02-2022  57.575210   

                   Light_Conditions           District Area  Longitude  \
0             Darkness - lights lit  Kensington and Chelsea  -0.209082   
1                          Daylight  Kensington and Chelsea  -0.173647   
2       Darkness - lighting unknown  Kensington and Chelsea  -0.210215   
3                          Daylight  Kensington and Chelsea  -0.202731   
4             Darkness - lights lit  Kensington and Chelsea  -0.192487   
...                             ...                     ...        ...   
660674                     Daylight                Highland  -3.467828   
660675       Darkness - no lighting                Highland  -3.809281   
660676                     Daylight                Highland  -3.862727   
660677       Darkness - no lighting                Highland  -3.823997   
660678                     Daylight                Highland  -3.895673   

        Number_of_Casualties  Number_of_Vehicles Road_Surface_Conditions  \
0                          1                   2                     Dry   
1                          1                   2             Wet or damp   
2                          1                   3                     Dry   
3                          1                   4                     Dry   
4                          1                   2                     Dry   
...                      ...                 ...                     ...   
660674                     2                   1                     Dry   
660675                     1                   1            Frost or ice   
660676                     1                   3            Frost or ice   
660677                     1                   2             Wet or damp   
660678                     1                   1             Wet or damp   

                 Road_Type Urban_or_Rural_Area     Weather_Conditions  \
0       Single carriageway               Urban     Fine no high winds   
1       Single carriageway               Urban  Raining no high winds   
2                      NaN               Urban                    NaN   
3       Single carriageway               Urban     Fine no high winds   
4                      NaN               Urban                    NaN   
...                    ...                 ...                    ...   
660674  Single carriageway               Rural     Fine no high winds   
660675  Single carriageway               Rural     Fine no high winds   
660676  Single carriageway               Rural     Fine no high winds   
660677  Single carriageway               Rural     Fine no high winds   
660678    Dual carriageway               Rural  Snowing no high winds   

                                Vehicle_Type  
0                                        Car  
1                                        Car  
2                      Taxi/Private hire car  
3       Bus or coach (17 or more pass seats)  
4                              Other vehicle  
...                                      ...  
660674                                   Car  
660675                                   Car  
660676                                   Car  
660677                 Motorcycle over 500cc  
660

In [5]:
uk.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype  
---  ------                   --------------   -----  
 0   Index                    660679 non-null  object 
 1   Accident_Severity        660679 non-null  object 
 2   Accident Date            660679 non-null  object 
 3   Latitude                 660654 non-null  float64
 4   Light_Conditions         660679 non-null  object 
 5   District Area            660679 non-null  object 
 6   Longitude                660653 non-null  float64
 7   Number_of_Casualties     660679 non-null  int64  
 8   Number_of_Vehicles       660679 non-null  int64  
 9   Road_Surface_Conditions  659953 non-null  object 
 10  Road_Type                656159 non-null  object 
 11  Urban_or_Rural_Area      660664 non-null  object 
 12  Weather_Conditions       646551 non-null  object 
 13  Vehicle_Type             660679 non-null  object 
dtypes: f

<h3>Changing the data types of the columns</h3>

In [6]:
uk.dtypes

Index                       object
Accident_Severity           object
Accident Date               object
Latitude                   float64
Light_Conditions            object
District Area               object
Longitude                  float64
Number_of_Casualties         int64
Number_of_Vehicles           int64
Road_Surface_Conditions     object
Road_Type                   object
Urban_or_Rural_Area         object
Weather_Conditions          object
Vehicle_Type                object
dtype: object

In [7]:
uk['Index'] = uk['Index'].astype('category')
uk['Accident_Severity'] = uk['Accident_Severity'].astype('category')
uk['Accident Date'] = uk['Accident Date'].astype('category')
uk['Light_Conditions'] = uk['Light_Conditions'].astype('category')
uk['District Area'] = uk['District Area'].astype('category')
uk['Road_Surface_Conditions'] = uk['Road_Surface_Conditions'].astype('category')
uk['Road_Type'] = uk['Road_Type'].astype('category')
uk['Urban_or_Rural_Area'] = uk['Urban_or_Rural_Area'].astype('category')
uk['Weather_Conditions'] = uk['Weather_Conditions'].astype('category')
uk['Vehicle_Type'] = uk['Vehicle_Type'].astype('category')

In [8]:
uk.dtypes

Index                      category
Accident_Severity          category
Accident Date              category
Latitude                    float64
Light_Conditions           category
District Area              category
Longitude                   float64
Number_of_Casualties          int64
Number_of_Vehicles            int64
Road_Surface_Conditions    category
Road_Type                  category
Urban_or_Rural_Area        category
Weather_Conditions         category
Vehicle_Type               category
dtype: object

<h3>Cleaning empty datas</h3>

In [9]:
uk.isna().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h5>Cleaning empty numerical datas</h5>

In [10]:
uk['Latitude'] = uk['Latitude'].fillna(uk['Latitude'].mean())
uk['Longitude'] = uk['Longitude'].fillna(uk['Longitude'].mean())

In [11]:
uk.isna().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                       0
Light_Conditions               0
District Area                  0
Longitude                      0
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h5>Cleaning non-numerical empty datas </h5>

In [12]:
uk['Road_Surface_Conditions'] = uk['Road_Surface_Conditions'].fillna(uk['Road_Surface_Conditions'].mode()[0])
uk['Road_Type'] = uk['Road_Type'].fillna(uk['Road_Type'].mode()[0])
uk['Urban_or_Rural_Area'] = uk['Urban_or_Rural_Area'].fillna(uk['Urban_or_Rural_Area'].mode()[0])
uk['Weather_Conditions'] = uk['Weather_Conditions'].fillna(uk['Weather_Conditions'].mode()[0])

In [13]:
uk.isna().sum()

Index                      0
Accident_Severity          0
Accident Date              0
Latitude                   0
Light_Conditions           0
District Area              0
Longitude                  0
Number_of_Casualties       0
Number_of_Vehicles         0
Road_Surface_Conditions    0
Road_Type                  0
Urban_or_Rural_Area        0
Weather_Conditions         0
Vehicle_Type               0
dtype: int64

<h1>Exploratory Data Analytics</h5>

<h1>1. What percentages do each accident severity represent on the accident report? and which severity had the highest percentage on the report.

In [14]:
uk['Accident_Severity'].unique()

['Serious', 'Slight', 'Fatal']
Categories (3, object): ['Fatal', 'Serious', 'Slight']

In [15]:
severity = uk['Accident_Severity']

In [16]:
Slight, Serious, Fatal = severity.value_counts()

<p>Percentage of Slight category</p>

In [17]:
(Slight / len(uk)) * 100

85.33660067899842

<p>Slight severity consumes <strong>85%</strong> of overall reports</p>

<p>Percentage of serious category</p>

In [18]:
(Serious / len(uk)) * 100

13.352475256516403

<p>Serious severity consumes <strong>13%</strong> of overall reports</p>

In [19]:
(Fatal / len(uk)) * 100

1.3109240644851736

<p>Fatal severity consumes <strong>1%</strong> of overall reports</p>

<h1>Insight # 1: With the calculation above, we found out that Serious accident severity consumes 85.33% of the overall reportts while Serious and Fatal have 13.35% and 1.31% respectively</h1>

<h1>2. Does the weather condition and vehicle type affect the number of casualties? If so, which weather condition and vehicle type recorded the highest number of casualties?

In [20]:
 pd.set_option('display.max_colwidth', 500)

In [21]:
weatherandvehicle = uk.groupby(['Weather_Conditions', "Vehicle_Type"])['Number_of_Casualties'].size()

In [22]:
uk['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       535013
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<h3>Breakdown</h3>

In [23]:
# Fine and no high winds
weatherandvehicle['Fine no high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      403324
Van / Goods 3.5 tonnes mgw or under       27603
Bus or coach (17 or more pass seats)      20963
Motorcycle over 500cc                     20793
Goods 7.5 tonnes mgw and over             13938
Motorcycle 125cc and under                12407
Taxi/Private hire car                     10733
Motorcycle over 125cc and up to 500cc      6228
Motorcycle 50cc and under                  6165
Goods over 3.5t. and under 7.5t            4950
Other vehicle                              4591
Minibus (8 - 16 passenger seats)           1588
Agricultural vehicle                       1555
Pedal cycle                                 167
Data missing or out of range                  5
Ridden horse                                  3
Name: Number_of_Casualties, dtype: int64

In [24]:
# raining no high winds
weatherandvehicle['Raining no high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      59940
Van / Goods 3.5 tonnes mgw or under       4203
Bus or coach (17 or more pass seats)      3182
Motorcycle over 500cc                     3135
Goods 7.5 tonnes mgw and over             2114
Motorcycle 125cc and under                1830
Taxi/Private hire car                     1622
Motorcycle 50cc and under                  897
Motorcycle over 125cc and up to 500cc      893
Goods over 3.5t. and under 7.5t            729
Other vehicle                              643
Minibus (8 - 16 passenger seats)           245
Agricultural vehicle                       244
Pedal cycle                                 18
Ridden horse                                 1
Data missing or out of range                 0
Name: Number_of_Casualties, dtype: int64

In [25]:
# raining+ high winds
weatherandvehicle['Raining + high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      7206
Van / Goods 3.5 tonnes mgw or under       508
Bus or coach (17 or more pass seats)      389
Motorcycle over 500cc                     354
Goods 7.5 tonnes mgw and over             275
Motorcycle 125cc and under                220
Taxi/Private hire car                     190
Motorcycle over 125cc and up to 500cc     117
Motorcycle 50cc and under                 107
Other vehicle                              99
Goods over 3.5t. and under 7.5t            83
Agricultural vehicle                       37
Minibus (8 - 16 passenger seats)           27
Pedal cycle                                 3
Data missing or out of range                0
Ridden horse                                0
Name: Number_of_Casualties, dtype: int64

In [26]:
# Fine + high winds
weatherandvehicle['Fine + high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      6463
Van / Goods 3.5 tonnes mgw or under       408
Bus or coach (17 or more pass seats)      350
Motorcycle over 500cc                     315
Goods 7.5 tonnes mgw and over             251
Taxi/Private hire car                     187
Motorcycle 125cc and under                180
Motorcycle 50cc and under                 106
Goods over 3.5t. and under 7.5t            94
Motorcycle over 125cc and up to 500cc      87
Other vehicle                              57
Minibus (8 - 16 passenger seats)           34
Agricultural vehicle                       20
Data missing or out of range                1
Pedal cycle                                 1
Ridden horse                                0
Name: Number_of_Casualties, dtype: int64

In [27]:
# snowing no high winds
weatherandvehicle['Snowing no high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      4748
Van / Goods 3.5 tonnes mgw or under       296
Motorcycle over 500cc                     254
Bus or coach (17 or more pass seats)      219
Goods 7.5 tonnes mgw and over             149
Motorcycle 125cc and under                146
Taxi/Private hire car                     136
Motorcycle 50cc and under                  75
Motorcycle over 125cc and up to 500cc      72
Goods over 3.5t. and under 7.5t            52
Other vehicle                              50
Agricultural vehicle                       22
Minibus (8 - 16 passenger seats)           16
Pedal cycle                                 3
Data missing or out of range                0
Ridden horse                                0
Name: Number_of_Casualties, dtype: int64

In [28]:
# Fog or mist
weatherandvehicle['Fog or mist'].sort_values(ascending = False)

Vehicle_Type
Car                                      2641
Van / Goods 3.5 tonnes mgw or under       192
Bus or coach (17 or more pass seats)      134
Motorcycle over 500cc                     118
Goods 7.5 tonnes mgw and over              93
Motorcycle 125cc and under                 82
Taxi/Private hire car                      78
Motorcycle 50cc and under                  50
Motorcycle over 125cc and up to 500cc      45
Other vehicle                              38
Goods over 3.5t. and under 7.5t            31
Minibus (8 - 16 passenger seats)           14
Agricultural vehicle                        9
Pedal cycle                                 3
Data missing or out of range                0
Ridden horse                                0
Name: Number_of_Casualties, dtype: int64

In [29]:
# Snowing + high winds
weatherandvehicle['Snowing + high winds'].sort_values(ascending = False)

Vehicle_Type
Car                                      677
Van / Goods 3.5 tonnes mgw or under       48
Motorcycle over 500cc                     33
Bus or coach (17 or more pass seats)      32
Goods 7.5 tonnes mgw and over             28
Taxi/Private hire car                     20
Motorcycle 125cc and under                16
Motorcycle over 125cc and up to 500cc     10
Motorcycle 50cc and under                  6
Goods over 3.5t. and under 7.5t            6
Other vehicle                              4
Agricultural vehicle                       3
Minibus (8 - 16 passenger seats)           2
Data missing or out of range               0
Ridden horse                               0
Pedal cycle                                0
Name: Number_of_Casualties, dtype: int64

<h1>Insight #2:</h1>
<h5>Car consistently dominated the vehucle type per casualties. It remained as the number one cause of accident regardlessof the weather condition. Surprisingly, contrary to the popular belief that wet roads causes more accidents, the data shows that 500000 of the overall report occured on a dry weather.

<h1>Question #3: Which is the safest? Urban or Rural area. Determine if the Light condition and Road type of an area determine the number of casualties it  will record. </h1>

In [30]:
uk.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 660679 entries, 0 to 660678
Data columns (total 14 columns):
 #   Column                   Non-Null Count   Dtype   
---  ------                   --------------   -----   
 0   Index                    660679 non-null  category
 1   Accident_Severity        660679 non-null  category
 2   Accident Date            660679 non-null  category
 3   Latitude                 660679 non-null  float64 
 4   Light_Conditions         660679 non-null  category
 5   District Area            660679 non-null  category
 6   Longitude                660679 non-null  float64 
 7   Number_of_Casualties     660679 non-null  int64   
 8   Number_of_Vehicles       660679 non-null  int64   
 9   Road_Surface_Conditions  660679 non-null  category
 10  Road_Type                660679 non-null  category
 11  Urban_or_Rural_Area      660679 non-null  category
 12  Weather_Conditions       660679 non-null  category
 13  Vehicle_Type             660679 non-null  ca

In [31]:
urbanorrural = uk.groupby(['Urban_or_Rural_Area','Light_Conditions','Road_Type'])['Number_of_Casualties'].count()

<h3>Breakdown</h3>

In [32]:
# Rural
urbanorrural['Rural']

Light_Conditions             Road_Type         
Darkness - lighting unknown  Dual carriageway         512
                             One way street            15
                             Roundabout               188
                             Single carriageway      1702
                             Slip road                 50
Darkness - lights lit        Dual carriageway        6556
                             One way street           182
                             Roundabout              3026
                             Single carriageway     14248
                             Slip road                683
Darkness - lights unlit      Dual carriageway         254
                             One way street             6
                             Roundabout                71
                             Single carriageway       611
                             Slip road                 19
Darkness - no lighting       Dual carriageway        7019
                             One way street            74
                             Roundabout               172
                             Single carriageway     27926
                             Slip road                326
Daylight                     Dual carriageway       34374
                             One way street           916
                             Roundabout             12088
                             Single carriageway    124756
                             Slip road               3216
Name: Number_of_Casualties, dtype: int64

In [33]:
urbanorrural['Urban']

Light_Conditions             Road_Type         
Darkness - lighting unknown  Dual carriageway         393
                             One way street           109
                             Roundabout               282
                             Single carriageway      3197
                             Slip road                 36
Darkness - lights lit        Dual carriageway       13686
                             One way street          3128
                             Roundabout              7501
                             Single carriageway     79691
                             Slip road                632
Darkness - lights unlit      Dual carriageway         211
                             One way street            52
                             Roundabout               118
                             Single carriageway      1186
                             Slip road                 15
Darkness - no lighting       Dual carriageway         441
                             One way street            29
                             Roundabout                52
                             Single carriageway      1367
                             Slip road                 31
Daylight                     Dual carriageway       35977
                             One way street          9048
                             Roundabout             20493
                             Single carriageway    241970
                             Slip road               2033
Name: Number_of_Casualties, dtype: int64

In [34]:
# Checking if Rural recorded more number of casualties than Urban
(urbanorrural['Rural'] > urbanorrural['Urban']).value_counts()

Number_of_Casualties
False    14
True     11
Name: count, dtype: int64

<p>We could say here that rural is safer than Urban, but we still need to compare the number of casualties each have</p>

In [35]:
urbanorrural['Rural'].sum()

np.int64(238990)

In [36]:
urbanorrural['Urban'].sum()

np.int64(421678)

<p>Now, let's determine which Road Type and Light condition from each area recorded the highest number of casualties

In [37]:
urbanorrural['Rural'].sort_values(ascending=False).head(n=5)

Light_Conditions        Road_Type         
Daylight                Single carriageway    124756
                        Dual carriageway       34374
Darkness - no lighting  Single carriageway     27926
Darkness - lights lit   Single carriageway     14248
Daylight                Roundabout             12088
Name: Number_of_Casualties, dtype: int64

In [38]:
urbanorrural['Urban'].sort_values(ascending=False).head(n=5)

Light_Conditions       Road_Type         
Daylight               Single carriageway    241970
Darkness - lights lit  Single carriageway     79691
Daylight               Dual carriageway       35977
                       Roundabout             20493
Darkness - lights lit  Dual carriageway       13686
Name: Number_of_Casualties, dtype: int64

<h1>Insight #3:</h1>
<p>Daylight recorded the highest number of casualties, this would mean that drivers were more careful at night that they don't take extra car during daytime. Also, single carriageway road types recorded the highest number of casualties

<h1>Question #4: Traffic is one of the main source of accidents, determine which district areas recorded an accident involving more than 5 cars</h1>

In [39]:
more_than_5 = uk[uk['Number_of_Casualties'] > 5]
uk.groupby(more_than_5['District Area'])['Number_of_Casualties'].count().sort_values(ascending=False)

District Area
Birmingham             70
Bradford               59
Liverpool              59
Leeds                  48
Manchester             43
                       ..
Bracknell Forest        0
Berwick-upon-Tweed      0
Bexley                  0
Wellingborough          0
West Dunbartonshire     0
Name: Number_of_Casualties, Length: 422, dtype: int64

<h1>Insight #4:</h1>
<p>Birmingham District recorded 70 casualties involving more than 5 cars</p>

<h1>Question #5: Since Birminham District has the highest number of casualties involving more than 5 cars, determine what type of road surface condition they have.</h1>

In [46]:
birmingham = uk[uk['District Area'] == 'Birmingham']
birmingham_road_surface = birmingham[birmingham['Number_of_Casualties'] > 5]['Road_Surface_Conditions']
birmingham_road_surface.value_counts()

Road_Surface_Conditions
Dry                     48
Wet or damp             20
Frost or ice             1
Snow                     1
Flood over 3cm. deep     0
Name: count, dtype: int64

<h1>Insight #5:</h1>
<p>Strengthening the conclusion that we had earlier wet surface is not the main source of accident</p>

<h1>Question #6: Which district are have the highest number of fatal accidents</h1>

In [50]:
Fatal_per_district = uk[uk['Accident_Severity'] == 'Fatal']['District Area'].value_counts()

In [52]:
Fatal_per_district.sort_values(ascending=False).head(n=10)

District Area
Birmingham                  105
Leeds                        93
Highland                     88
East Riding of Yorkshire     85
Bradford                     71
Aberdeenshire                66
Powys                        59
Doncaster                    56
Wakefield                    56
Herefordshire, County of     51
Name: count, dtype: int64

<h1>Insight #6:</h1>
<p>Birmingham district have the highest number of Fatal accidents</p>

<h1>Question #7: What is the common vehicle types involved in Fatal Accidents?</h1>

In [53]:
Vehicle_fatal = uk[uk['Accident_Severity'] == 'Fatal']['Vehicle_Type'].value_counts()

In [54]:
Vehicle_fatal.sort_values(ascending=False).head(n=10)

Vehicle_Type
Car                                      6577
Van / Goods 3.5 tonnes mgw or under       467
Motorcycle over 500cc                     339
Bus or coach (17 or more pass seats)      325
Goods 7.5 tonnes mgw and over             216
Motorcycle 125cc and under                189
Taxi/Private hire car                     155
Motorcycle over 125cc and up to 500cc     105
Motorcycle 50cc and under                  95
Other vehicle                              70
Name: count, dtype: int64

<h1>Insight #7:</h1>
<p>Car remains the number one vehicle involved on fatal accidents</p>

<h1>Question #8: Is there a relationship between latitude and longitude and the number of casualties?</h1>

In [62]:
uk['Number_of_Casualties'].corr(uk['Latitude'])

np.float64(0.03220068662590627)

In [63]:
uk['Number_of_Casualties'].corr(uk['Longitude'])

np.float64(-0.04040564578845446)

<h1>Insight #8:</h1>
<p>There's no relation between longitude, latitude, and number of casualties</p>

<h1>Question #9: Is there a correlation between the number of vehicles involved had the number of casualties</h1>

In [64]:
uk['Number_of_Casualties'].corr(uk['Number_of_Vehicles'])

np.float64(0.22888886126927635)

<h1>Insight #9: No, there is no correlation between Number of vehicles involved and the number of casualties</h1>

<h1>Question #10: Which road surface conditions are prevalent during serious accidents?</h1>

In [70]:
serious_accident = uk[uk['Accident_Severity'] == 'Serious']
surface_condition_serious = serious_accident['Road_Surface_Conditions'].value_counts()
surface_condition_serious.sort_values(ascending=False)

Road_Surface_Conditions
Dry                     61708
Wet or damp             23785
Frost or ice             2007
Snow                      565
Flood over 3cm. deep      152
Name: count, dtype: int64

<h1>Insight #10:</h1>
<p>Dry surface are prevalent during serious accidents</h1>

<h1>Insight #11: What are the most common combinations of weather conditions and light conditions during accidents?</h1>

In [73]:
weather_and_light = uk.groupby(['Weather_Conditions','Light_Conditions'])['Number_of_Casualties'].count()
weather_and_light.sort_values(ascending=False).head(n=10)

Weather_Conditions     Light_Conditions           
Fine no high winds     Daylight                       408726
                       Darkness - lights lit           93958
Raining no high winds  Daylight                        49742
Fine no high winds     Darkness - no lighting          25251
Raining no high winds  Darkness - lights lit           22666
Other                  Daylight                        10103
Raining no high winds  Darkness - no lighting           6208
Fine + high winds      Daylight                         5796
Fine no high winds     Darkness - lighting unknown      5333
Raining + high winds   Daylight                         4942
Name: Number_of_Casualties, dtype: int64

<h1>Insight #11:</h1>
<p>Fine no high winds and daylight are most common combination if weather and light conditions during accidents</p>

<h1>Question #12: What are the most frequent 'Vehicle_Type' involved in accidents under 'Darkness - lights lit' conditions?</h1>

In [74]:
darkness_lights_lit = uk[uk['Light_Conditions'] == 'Darkness - lights lit']
darkness_lights_lit['Vehicle_Type'].value_counts()

Vehicle_Type
Car                                      96994
Van / Goods 3.5 tonnes mgw or under       6806
Motorcycle over 500cc                     5176
Bus or coach (17 or more pass seats)      5142
Goods 7.5 tonnes mgw and over             3440
Motorcycle 125cc and under                3074
Taxi/Private hire car                     2658
Motorcycle 50cc and under                 1494
Motorcycle over 125cc and up to 500cc     1480
Goods over 3.5t. and under 7.5t           1192
Other vehicle                             1126
Agricultural vehicle                       365
Minibus (8 - 16 passenger seats)           347
Pedal cycle                                 39
Data missing or out of range                 1
Ridden horse                                 1
Name: count, dtype: int64

<h1>Insight #12: </h1>
<p>Car is the most common vehicle under Darkness - Lights lit condition</p>